gaia_tasks/inputs.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc. and Epidemico Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+import json
+import os
+import tempfile
+from base64 import b64encode
+
+import fiona
+import geopandas
+import requests
+# from girder.plugins.minerva.utility.minerva_utility import decryptCredentials
+from six import StringIO
+
+from gaia.core import GaiaException
+from gaia.filters import filter_pandas
+from gaia.inputs import GaiaIO
+from gaia import formats, types
+from girder.utility import config
+from girder.constants import PACKAGE_DIR
+import girder_client
+
+
+class MinervaVectorIO(GaiaIO):
+    """
+    Interface to Minerva item geojson
+    """
+
+    type = types.VECTOR
+    default_output = formats.JSON
+
+    def __init__(self, item_id=None, token=None, name='gaia_result.json',
+                 uri='', **kwargs):
+        """
+        Read and write GeoJSON data to/from Girder
+        :param item_id: Item id to read/write from/to
+        :param uri: location of temporary file
+        :param kwargs: Other keyword arguments
+        """
+
+        self.id = item_id
+        self.token = token
+        if uri:
+            self.uri = uri
+        else:
+            tmpdir = tempfile.mkdtemp()
+            self.uri = tempfile.mkstemp(suffix='.json', dir=tmpdir)[1]
+        self.filename = name
+        girderHost = config.getConfig().get('minerva', {}).get('worker_girder_host_name', 'localhost')
+        girderScheme = config.getConfig().get('minerva', {}).get('worker_girder_host_scheme', 'http')
+
+        girderPort = config.getConfig()['server.socket_port']
```

Comment:
If I understand the code correctly, this will run in worker, which doesn't necessarily run on the same machine as girder.  You have to pass this information from the producer side (girder).

girder_worker will bind a correct girder client instance to your task for you to use.  See for example [this integration test](https://github.com/girder/girder_worker/blob/3c6b4774f8477d4107e3667885f64b901f90eb02/tests/integration/common_tasks/common_tasks/test_tasks/girder_client.py).

Reply:
I felt the way I am providing the config is unorthodox, but the current implementation of girder.utility.config is able to get the config no matter when it's running as girder or girder_worker.
But I do see the Bind injected grider_client is better. I'll change the code to use that. Do you know if there is a sample code that I can refer to on "passing this information from the producer side"

Reply:
I think @jbeezley  point was,  the config may not be available because girder worker may (and in fact will likely) be running on a totally different computer.  

In general when developing with girder worker it is a good idea to at least be running girder worker in a separate virtualenv from girder to help surface some (although not all) of these issues

Reply:
@kotfic, Thank you for your comments, totally get what you and @jbeezley are saying. At this moment I copied the config to the worker instance/container, so it would work. However, I am aware this is not how we should use the new girder-worker. 
As people discussed in Gumbo meeting, I'll learn the proper way to use girder worker (with another project) and come back to this.

Reply:
After our conversation, it occurred to me that it would be better and easier to understand if you deployed this information into the [worker configuration file](https://github.com/girder/girder_worker/blob/master/girder_worker/worker.dist.cfg) instead.  You should be able to get the config object at `girder_worker.config`.

Reply:
@jbeezley I see. I pushed a commit to use girder worker config instead. I also pushed a commit to the related PR in minerva-docker.